src/components/Tokens/TokenDetails/TransactionsTable.tsx

```diff
@@ -0,0 +1,175 @@
+          <TableCell alignRight>
+            <ThemedText.BodySecondary>${referenceToken.symbol}</ThemedText.BodySecondary>
+          </TableCell>
+        ),
+        cell: (inputTokenAmount) => (
+          <TableCell alignRight>
+            <ThemedText.BodyPrimary>
+              {formatNumber({ input: inputTokenAmount.getValue(), type: NumberType.TokenTx })}
+            </ThemedText.BodyPrimary>
+          </TableCell>
+        ),
+      }
+    ),
+    columnHelper.accessor(
+      (row) => {
+        const nonReferenceSwapLeg =
+          row.input.address.toLowerCase() === referenceToken.address.toLowerCase() ? row.output : row.input
+        return `${formatNumber({
+          input: nonReferenceSwapLeg.amount,
+          type: NumberType.TokenTx,
+        })} ${nonReferenceSwapLeg.symbol}`
+      },
+      {
+        id: 'non-reference-amount',
+        header: () => (
+          <TableCell alignRight>
+            <ThemedText.BodySecondary>
+              <Trans>For</Trans>
+            </ThemedText.BodySecondary>
+          </TableCell>
+        ),
+        cell: (swapOutput) => (
+          <TableCell alignRight>
+            <ThemedText.BodyPrimary>{swapOutput.getValue()}</ThemedText.BodyPrimary>
+          </TableCell>
+        ),
+      }
+    ),
+    columnHelper.accessor((row) => row.usdValue, {
+      id: 'usd-value',
+      header: () => (
+        <TableCell alignRight>
+          <ThemedText.BodySecondary>
+            <Trans>USD</Trans>
+          </ThemedText.BodySecondary>
+        </TableCell>
+      ),
+      cell: (usd) => (
+        <TableCell alignRight>
+          <ThemedText.BodyPrimary>
+            {formatNumber({ input: usd.getValue(), type: NumberType.FiatTokenPrice })}
+          </ThemedText.BodyPrimary>
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor((row) => row.makerAddress, {
+      id: 'maker-address',
+      header: () => (
+        <TableCell alignRight>
+          <ThemedText.BodySecondary>
+            <Trans>Maker</Trans>
+          </ThemedText.BodySecondary>
+        </TableCell>
+      ),
+      cell: (makerAddress) => (
+        <TableCell alignRight>
+          <ThemedText.BodyPrimary>{shortenAddress(makerAddress.getValue(), 0)}</ThemedText.BodyPrimary>
+        </TableCell>
+      ),
+    }),
+    columnHelper.accessor((row) => getExplorerLink(chainId, row.hash, ExplorerDataType.TRANSACTION), {
+      id: 'etherscan-link',
+      header: () => (
+        <TableCell alignRight>
+          <ThemedText.BodySecondary>
+            <Trans>Tx</Trans>
+          </ThemedText.BodySecondary>
+        </TableCell>
+      ),
+      cell: (explorerLink) => (
+        <TableCell alignRight>
+          <StyledExternalLink href={explorerLink.getValue()} data-testid={explorerLink.getValue()}>
+            <ExternalLinkIcon size="16px" stroke={theme.neutral1} />
+          </StyledExternalLink>
+        </TableCell>
+      ),
+    }),
+  ]
+  // TODO: use live data
```

Comment:
add Linear story to comment, alson in `#proj-interface-info` I added the base for adding token tx query